examples/ivr_outbound_dialer.py

```diff
@@ -0,0 +1,374 @@
+import switchy
+import threading
+from switchy.marks import event_callback
+from switchy.apps.players import TonePlay
+from switchy import get_originator
+
+# Enable logging to stderr
+log = switchy.utils.log_to_stderr()
+# Debug levels: INFO in production, DEBUG in devel
+log.setLevel("INFO")
+
+# Specify FreeSWITCH or Sangoma NSG IP information
+# In this example the sample app is running on 
+# Sangoma NSG appliance hence the use of local address
+host = "127.0.0.1"
+port = 8821
+
+
+# IVR Call Logic
+# This is the main IVR logic
+# All custom development should be done here.
+# Variables
+#   self.<variables>       are global in nature.
+#   call.vars.['var_name'] should be used for per call info and state
+#
+class IVRCallLogic(object):
+
+    def prepost(self, client, listener):
+        # Get the install directory of FreeSWITCH or Sangoma NSG and append recording to it
+        # By default FreeSWITCH resides in /usr/local/freeswitch, NSG resides in /usr/local/nsg
+        self.recdir = client.cmd('global_getvar base_dir')
+        self.recdir += "/recording"
+        log.info("Setting recording dir to '{}" . format(self.recdir))
+        
+        # Get the install directory of NSG and append sounds to it
+        self.sound_dir = client.cmd('global_getvar base_dir')
+        self.sound_dir += "/sounds"
+        log.info("Setting sounds dir to '{}" . format(self.sound_dir))
+
+        self.stereo=False
+
+        # mod_sndfile module is a must in order to play prompts
+        # Example of how to execute FreeSWITCH/NSG commands as from the CLI
+        try:
+            client.cmd('load mod_sndfile')
+        except:
+            pass
+   
+        # Example of how to use a Timer to measure elapsed time
+        # This example does not use it, but its use is commented out and
+        # left for demonstration purposes 
+        self.dtmf_timeout = switchy.utils.Timer()
+        self.dtmf_timeout_period = 3.0
+
+		# process originate events the same as create events since on some versions 
+        # of FS originate events will arrive first (which is super stupid)
+        listener._handlers.pop("CHANNEL_ORIGINATE")
+        listener.add_handler('CHANNEL_ORIGINATE', listener._handle_create)
+
+    @event_callback('CHANNEL_PARK')
+    def on_park(self, sess):
+        # Example of how to answer an inbound call
+        # This application demonstrates outbound dialing
+        if sess.is_inbound():
+            sess.answer()
+    
+    @event_callback("CHANNEL_ANSWER")
+    def on_answer(self, sess):
+        call = sess.call
+
+        # This application does not deal with inbound calls.
+        if sess.is_inbound():
+            pass
+
+        # Outbound call hsa just been answered
+        # Developer would start the introductory IVR message 
+        if sess.is_outbound():
+
+            # Start recording a call
+            call.vars['record']=True
+            self.rec_filename = '{}/callee_{}.wav'.format(self.recdir, sess.uuid)
```

Comment:
Avoid the use of the "global" instance variable since you reassign it on every call anyway.
```python
sess.start_record(
    '{}/callee_{}.wav'.format(self.recdir, sess.uuid),
    stereo=self.stereo
)
```
Should do it.